src/chapters/setup-intro-data-types/alternate-code-practice.rst

```diff
@@ -0,0 +1,406 @@
+
+   .. figure:: figures/TempConverter-class.png
+      :alt: New TempConverter class.
+
+#. Cut and paste the following code into the editor. We will analyze the
+   different statements in a moment.
+
+   .. sourcecode:: java
+      :linenos:
+
+      import java.util.Scanner;
+
+      public class TempConverter {
+         public static void main(String[] args) {
+            double fahrenheit;
+            double celsius;
+            Scanner input;
+
+            input = new Scanner(System.in);
+            System.out.println("Enter the temperature in Fahrenheit: ");
+            fahrenheit = input.nextDouble();
+            input.close();
+
+            celsius = (fahrenheit - 32) * 5/9;
+            System.out.println("The temperature in Celsius is: " + celsius + "°C");
+         }
+      }
+
+#. Build and run the program to verify that it works. Entering a Fahrenheit
+   temperature of ``212`` yields the result, ``The temperature in Celsius is:
+   100.0°C``.
+
+There are several new concepts introduced in this example. We will look
+at them in the following order:
+
+#. The ``import`` statement
+#. Declaring variables
+#. Collecting input with the Scanner class
+
+``import``
+-----------
+
+The ``import`` statement in Java allows us to access the class, methods, and
+data stored a different file. ``import`` tells the compiler that we are going
+to use a shortened version of the class name. In this example, we are going to
+use the class ``java.util.Scanner``, but we can refer to it as just
+``Scanner``. We could use the ``java.util.Scanner`` class without any problem
+and without any import statement, provided that we always referred to it by its
+full name.
+
+In Java, you can use any class that is available WITHOUT having to import the
+class, but you must adhere to two very important conditions:
+
+#. The ``javac`` and ``java`` commands must know that the class exists.
+#. You must use the full name of the class.
+
+.. admonition:: Try It
+
+   In the sample code, remove the ``import`` statement in line 1, and change
+   ``Scanner`` on lines 7 & 9 to ``java.util.Scanner``. The program should
+   still compile and run.
+
+The class naming system in Java is very hierarchical. The *full* name of the
+``Scanner`` class is really ``java.util.Scanner``. You can think of this name
+as having two parts:
+
+#. ``java.util`` is called the *package*,
+#. ``Scanner`` is the class name.
+
+We’ll talk more about the class naming system a bit later.
+
+Java Knows Which Classes Exist, Kinda
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+How do the ``java`` and ``javac`` commands know that certain classes
+exist? We have these rules:
+
+#. Java knows about all the classes you define in the ``.java`` and ``.class``
+   files in your current working directory.
+#. Java knows about all the classes that get shipped with the software.
+#. Java knows about all the classes included in your ``CLASSPATH`` environment
```

Comment:
can we consider leaving this bullet point unmentioned for now? these are kind of heftier ideas they dont really need to grapple with yet when learning the basic syntax of the language

Reply:
Removed.  Thanks.